### ISSUE TITLE
Develop `usa-banner` alpha web component

### ISSUE THREAD

**Issue by heymatthenry** [uswds/uswds-next#1]
One of the first components we'll recreate as a Web Component is the `usa-banner` (uswds/uswds#5706).

## Necessary Tasks

These are the features that should constitute MVP for the pre-release version, since they will help us get our arms around the architectural choices we'll have to make in developing further components.

- [ ] Add `vitest` unit tests for component (WIP `vitest` setup is currently happening in #8)
- [ ] Add i18n support
- [ ] Ensure feature parity with existing banner component
- [ ] Ensure consistency of visual design with existing banner component
- [ ] Evaluate component's accessibility shortcomings and document them
- [ ] Add a Storybook story for the component (Depends on #6)

## Stretch Goals

All of these will need to happen at some point, but don't necessarily have to happen for the pre-release version

- [ ] Figure out how to package the component by itself for distribution (Depends on #5)
- [ ] Remediate any accessibility bugs

**Comment by humancompanion-usds** [comment]
FWIW, the VA does have a [Web Component implementation of the official usa banner](https://github.com/department-of-veterans-affairs/component-library/tree/main/packages/web-components/src/components/va-official-gov-banner). I hesitate to even mention it here given that I know we'll be a part of the working group providing feedback on Web Components, but just in case there is anything of use to you in seeing that implementation, there it is. We also tackled [i18n support which you can see in our Storybook](https://design.va.gov/storybook/?path=/docs/components-va-official-gov-banner--default#internationalization). I know there are many decisions to be made, but just in case there is anything in code that you can crib from us, by all means please do so.

**Comment by heymatthenry** [comment]
@humancompanion-usds Thank you! I'm sure we'll be cribbing heavily from the work y'all did. It's been very helpful to be able to refer to your implementations to get us thinking about the various levers available to us for customization, extensibility, etc. Looking forward to chatting more as the working group moves forward!
